tensorflow/c/eager/dlpack.cc

```diff
@@ -0,0 +1,342 @@
+#include "tensorflow/c/eager/dlpack.h"
+#include "include/dlpack/dlpack.h"  // TF:dlpack
+#include "tensorflow/c/eager/c_api_internal.h"
+#include "tensorflow/c/tf_status_helper.h"
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/platform/casts.h"
+
+#include "tensorflow/core/framework/tensor_reference.h"
+#include "tensorflow/core/platform/logging.h"
+
+namespace tensorflow {
+
+using tensorflow::Tensor;
+using tensorflow::TensorHandleInterface;
+
+namespace {
+
+struct TFDLMTensor {
+  TensorReference* handle;
+  DLManagedTensor tensor;
+};
+
+TensorHandle* GetTensorHandleFromTFEHandle(TFE_TensorHandle* h,
+                                           TF_Status* status) {
+  if (h == nullptr || !h->handle->IsValid(&status->status)) {
+    status->status = tensorflow::errors::InvalidArgument(
+        "The passed in handle is a nullptr");
+    return nullptr;
+  }
+  tensorflow::TensorHandle* handle =
+      tensorflow::down_cast<tensorflow::TensorHandleInterface*>(h->handle.get())
+          ->Handle();
+
+  if (handle->IsRemote()) {
+    status->status = tensorflow::errors::InvalidArgument(
+        "TFE_TensorHandleDevicePointer may not be called on a remote tensor "
```

Comment:
Probably different error message? (as in, I think we hit here without calling TFE_TensorHandleDevicePointer).

Same below.